kernel/include/kswap.h

```diff
@@ -106,6 +106,8 @@ static ALWAYS_INLINE unsigned int do_swap(unsigned int key,
 		z_spin_lock_set_owner(&sched_spinlock);
 #endif
 
+		arch_cohere_stacks(old_thread, NULL, new_thread);
+
```

Comment:
indeed, this fixes the regression, introduced by https://github.com/zephyrproject-rtos/zephyr/commit/6b84ab3830500224299c30f72c90c30bd4d9382d thanks!